toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/utilitynetwork/PercentAlongControl.kt

```diff
@@ -80,9 +82,10 @@ internal fun PercentAlongControl(
                 modifier = Modifier.fillMaxWidth()
             )
             Slider(
-                value = value,
+                value = value.toFloat(),
                 onValueChange = {
-                    value = it
+                    val decimal = BigDecimal(it.toString()).setScale(2, RoundingMode.DOWN)
```

Comment:
we can do this just by using arithmetic,
```suggestion
                    val decimal = kotlin.math.floor(it * 100) / 100
```
It’s efficient and avoids object allocation.

Reply:
I'm not sure this would work. The `floor` gives you the value as a `Float` but we need a `Double`. Converting this to a `Double` will result in the same precision/rounding problem as before.

Reply:
@puneet-pdx This gave me another idea though, we can just use implicit type conversion to a double like so, which would be more efficient as you noted.
```
value = (it * 100).toInt() / 100.0
```

Reply:
I tested the logic out and I was getting the same values for value, with what you had and `math.floor` approach

```Kotlin
val decimal = BigDecimal(it.toString()).setScale(2, RoundingMode.DOWN)
                    val decimal2 = kotlin.math.floor(it * 100) / 100
                    Log.i("PercentAlongControl", "Slider value changed to it ${it.toDouble()}")
                    Log.i("PercentAlongControl", "Slider value changed to decimal - $decimal")
                    Log.i("PercentAlongControl", "Slider value changed to decimal2 - $decimal2")
                    value = decimal.toDouble()
```

>I'm not sure this would work. The floor gives you the value as a Float but we need a Double. Converting this to a Double will result in the same precision/rounding problem as before.

I am not sure what you mean here, we are converting the value toDouble with the BigDecimal too

Reply:
> I am not sure what you mean here, we are converting the value toDouble with the BigDecimal too

For ex, if the value `it` from the slider is `0.63167262...`

`decimal2` is of type `Float` here. The api expects a `Double`. When you convert that into a double `decimal2.toDouble()` you might get something close to `0.6300009` due to binary conversion and precision. It will never really be `0.63`.

What we actually want is a double with a value of `0.63` which is exactly what the `BigDecimal` gives us.

Reply:
> When you convert that into a double decimal2.toDouble() you might get something close to 0.6300009 due to binary conversion and precision. It will never really be 0.63.

This was the actual initial issue btw - https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/blob/0666e2512aedabae29116060a744518a47601c47/toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/utilitynetwork/AddAssociationFromSourceViewModel.kt#L503